### PR TITLE
Fix fregrid conserve2 issues for stretched grids

### PR DESCRIPTION
**Pull request by nikizadehgfdl**
- Closes issue #180
- The centroid latitude calculation in ploy_ctrlat() needs a small fix similar to the fix in issue #45 for grid cells staddling the South pole but not including it as a vertex.
- Added/corrected comments in poly_area().
- Cleaned up poly_area() of unneeded code.

### CODE REVIEW

File: tools/libfrencutils/create_xgrid.c [hunk 1]
```diff
@@ -2100,13 +2099,57 @@ double poly_ctrlat(const double x[], const double y[], int n)
     avg_y = (lat1+lat2)*0.5;
     if      (dx==0.0) continue;
     if(dx > M_PI)  dx = dx - 2.0*M_PI;
-    if(dx < -M_PI) dx = dx + 2.0*M_PI;
+    if(dx <= -M_PI) dx = dx + 2.0*M_PI; // flip sign for dx=-pi to fix huge value see comments in function poly_area
```
Comment (nikizadehgfdl) [inline tools/libfrencutils/create_xgrid.c:2102]: The addition of  "=" sign is the main bug fix for issue #180 . The rest of this update is adding/fixing comments.

File: tools/libfrencutils/create_xgrid.c [hunk 2]
```diff
@@ -2100,13 +2099,57 @@ double poly_ctrlat(const double x[], const double y[], int n)
...
 
     if ( fabs(hdy)< SMALL_VALUE ) /* cheap area calculation along latitude */
       ctrlat -= dx*(2*cos(avg_y) + lat2*sin(avg_y) - cos(lat1) );
     else
       ctrlat -= dx*( (sin(hdy)/hdy)*(2*cos(avg_y) + lat2*sin(avg_y)) - cos(lat1) );
```
Comment (nikizadehgfdl) [inline tools/libfrencutils/create_xgrid.c:2107]: This formula for is a black box to me. All I know at this point is that it supposedly produces (centroid latitude) * (area)  , some kind of moment of inertia for the polygon on the sphere. It is later divided by the area in fregrid before being used as the centroid and hence has to have the same fix for sides passing thru poles as in poly_area().  I guess the derivation is along the lines I sketched for poly_area, but I have not been able to derive it or find any reference for it yet. If you know a reference or the derivation please add it here.

File: tools/libfrencutils/create_xgrid.c [hunk 3]
```diff
@@ -2100,13 +2099,57 @@ double poly_ctrlat(const double x[], const double y[], int n)
...
   }
+  if(fabs(ctrlat) > HPI) printf("WARNING poly_ctrlat: Large values for ctrlat: %19.15f\n", ctrlat);
+  return (ctrlat*RADIUS*RADIUS);
+}; /* poly_ctrlat */
+/*An alternate implementation of poly_ctrlat for future developments. Under construction.*/
+double poly_ctrlat2(const double x[], const double y[], int n)
```
Comment (nikizadehgfdl) [inline tools/libfrencutils/create_xgrid.c:2113]: An alternate implementation for later use. This is not used anywhere.

File: tools/libfrencutils/create_xgrid.c [hunk 4]
```diff
@@ -3041,6 +3084,10 @@ int main(int argc, char* argv[])
 
       n_out = clip_2dx2d(lon1_in, lat1_in, n1_in, lon2_in, lat2_in, n2_in, lon_out, lat_out);
 
+      n1_in = fix_lon(lon1_in, lat1_in, n1_in, M_PI);
```
Comment (nikizadehgfdl) [inline tools/libfrencutils/create_xgrid.c:3087]: fix_lon has to be applied  before poly_area to take care of the poles.

File: tools/libfrencutils/create_xgrid.c (hunk 2 above)
Comment (nikizadehgfdl) [reply tools/libfrencutils/create_xgrid.c:2107]: Nevermind, I found the derivation and the reference. It's Phil Jones's 1998 again. I'll add comments to the code.

**Review by ngs333 — APPROVED** [review]

**Pull request merged**